drivers/ieee802154/ieee802154_kw41z.c

```diff
@@ -24,12 +24,133 @@
 
 #include "fsl_xcvr.h"
 
+/*
+ * KW41Z M0 has Micro Trace Buffer support (MTB). The MTB is a hardware
+ * program flow tracing that traces the branch points of execution.
+ * Where this proves useful is when a hard fault occurs in the M0 you
+ * can have a breakpoint in the __reserved function (reset) and when
+ * the breakpoint happens look at the contents of the mtb buffer at
+ * the addresses traced.
+ *
+ * In gdb use: 'info line *<addr>'
+ *
+ */
+/*#define KW41Z_DEBUG_MTB_ENABLED   1*/
```

Comment:
Is there a reason why this is not enabled by default, the reason for that could be documented here.

Reply:
This is for debug if you get a fatal fault crash in the system. My intent is to propose a separate and global way of having this for platform level support and not specific to KW41Z.

Reply:
> My intent is to propose a separate and global way of having this for platform level support and not specific to KW41Z.

Definitely +1, and that's one of the things which can and should be separated from this commit (ideally, to a separate PR).

Reply:
Definitely a separate PR because MTB is not a KW41 specific debug item. I want to propose it as a debug feature for Zephyr moved to the platform level for ARM cores that support it. But since I needed it to debug some things in KW41, I put it in the driver code because I didn't have enough time to research how and where it should be for a platform level capability.